File: monitoring-backend/src/main/java/com/performance/monitoring/models/Device.java
package com.performance.monitoring.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "Devices")
public class Device {
    @Id
    private String id;

    @Indexed(unique = true)
    private String name;
    private String ip;
    private int port;
    private String organizationId;

    public Device(String id, String name, String ip, int port, String organizationId) {
        this.id = id;   
        this.name = name;
        this.ip = ip;
        this.port = port;
        this.organizationId = organizationId;
    }

    public String getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(String organizationId) {
        this.organizationId = organizationId;
    }
    
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIp() {
        return ip;
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public int getPort() {
        return port;
    }

    public void setPort(int port) {
        this.port = port;
    }
}

File: monitoring-frontend/src/models/device.ts
export class Device {
    id: string;
    name: string;
    ip: string;
    port: number;
    organizationId: string

    constructor(id: string, name: string, ip: string, port: number, organizationId: string) {
        this.id = id;
        this.name = name;
        this.ip = ip;   
        this.port = port;
        this.organizationId = organizationId;
    }
}

export class DeviceList {
    devices: Device[];

    constructor(devices: Device[]) {
        this.devices = devices;
    }
}